src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -494,69 +505,61 @@ static UCS_CLASS_INIT_FUNC(uct_cuda_copy_iface_t, uct_md_h md, uct_worker_h work
     self->config.bandwidth       = config->bandwidth;
     UCS_STATIC_BITMAP_RESET_ALL(&self->streams_to_sync);
 
-    ucs_mpool_params_reset(&mp_params);
-    mp_params.elem_size       = sizeof(uct_cuda_copy_event_desc_t);
-    mp_params.elems_per_chunk = 128;
-    mp_params.max_elems       = self->config.max_cuda_events;
-    mp_params.ops             = &uct_cuda_copy_event_desc_mpool_ops;
-    mp_params.name            = "CUDA EVENT objects";
-    status = ucs_mpool_init(&mp_params, &self->cuda_event_desc);
-    if (UCS_OK != status) {
-        ucs_error("mpool creation failed");
-        return UCS_ERR_IO_ERROR;
-    }
+    kh_init_inplace(cuda_copy_ctx_rscs, &self->ctx_rscs);
 
     ucs_queue_head_init(&self->active_queue);
 
+    return UCS_OK;
+}
+
+static void uct_cuda_copy_stream_destroy(CUstream *stream, int valid_ctx)
+{
+    if ((*stream == NULL) || (!valid_ctx)) {
```

Comment:
1. extra ( )
2. we probably want to set *stream=NULL also is valid_ctx = 0. or not set it at all since ctx_rsc is destroyed
3. suggest rename stream to stream_p